geode-membership/src/integrationTest/java/org/apache/geode/distributed/internal/membership/gms/MembershipIntegrationTest.java

```diff
@@ -272,6 +276,65 @@ public void secondMembershipPausesForLocatorWaitTime()
     stop(coordinatorLocator, lateJoiningLocator);
   }
 
+  @Test
+  public void locatorsStopWaitingForLocatorWaitTimeIfAllLocatorsContacted()
+      throws IOException, MemberStartupException, InterruptedException, TimeoutException,
+      ExecutionException {
+
+    final Supplier<ExecutorService> executorServiceSupplier =
+        () -> LoggingExecutors.newCachedThreadPool("membership", false);
+
+    int[] locatorPorts = AvailablePortHelper.getRandomAvailableTCPPorts(2);
+
+    int locatorWaitTime = (int) Duration.ofMinutes(5).getSeconds();
+    final MembershipConfig config =
+        createMembershipConfig(true, locatorWaitTime, locatorPorts[0], locatorPorts[1]);
+
+    CompletableFuture<Membership<MemberIdentifier>> createMembership0 =
+        launchLocator(executorServiceSupplier, locatorPorts[0], config);
+
+    // Assert that membership 0 is waiting for the other locator to start
+    Thread.sleep(5000);
+    assertThat(createMembership0.getNow(null)).isNull();
+
+    CompletableFuture<Membership<MemberIdentifier>> createMembership1 =
+        launchLocator(executorServiceSupplier, locatorPorts[1], config);
+
+    // Make sure the members are created in less than the locator-wait-time
+    Membership<MemberIdentifier> membership0 = createMembership0.get(2, TimeUnit.MINUTES);
+    Membership<MemberIdentifier> membership1 = createMembership1.get(2, TimeUnit.MINUTES);
+
+    // Make sure the members see each other in the view
+    assertThat(membership0.getView().getMembers()).hasSize(2);
+    assertThat(membership1.getView().getMembers()).hasSize(2);
+  }
```

Comment:
This is a good test and it's acceptable as-is. I wonder though, if you'd be willing to entertain reducing run-time and increasing determinism by by injecting the (millisecond) time keeper and sleeper into the locator, the way it's done for `PrimaryHandler`?

Here is the PR from August that added that injection: https://github.com/apache/geode/pull/5422

As of that PR (and now on `develop`) there are a couple functional interfaces defined in `PrimaryHander` that could be hoisted a little higher and used in `MembershipLocator` for your purposes, i.e. `Sleeper`, `MillisecondProvider`.

With those injected, this test could control time similar to how `PrimaryHandlerTest` does.

Reply:
Based on your comment I started down this road. But this is a test of the whole membership system, so this will require injecting virtual time into a whole lot of classes, including classes that are doing things like `wait(sometimeout)` etc. I feel like this needs a little bit more time and careful development to get right, and more than I want to shoehorn into this PR.

Reply:
Ok @upthewaterspout. After this PR merges I'll write a ticket to make this improvement. Thanks for trying it though!

Reply:
Here's that new ticket: https://issues.apache.org/jira/browse/GEODE-8589